pkg/controller/servicecatalog/provisionedproduct/setup.go

```diff
@@ -0,0 +1,265 @@
+	provisionedProductOutputs []*svcsdk.RecordOutput
+}
+
+func (c *custom) isUpToDate(cr *svcapitypes.ProvisionedProduct, resp *svcsdk.DescribeProvisionedProductOutput) (bool, error) {
+	if aws.StringValue(cr.Status.AtProvider.Status) == string(svcapitypes.ProvisionedProductStatus_SDK_UNDER_CHANGE) {
+		return true, nil
+	}
+	getPPInput := &svcsdk.GetProvisionedProductOutputsInput{ProvisionedProductId: resp.ProvisionedProductDetail.Id}
+	getPPOutput, err := c.client.GetProvisionedProductOutputs(getPPInput)
+	if err != nil {
+		return true, err
+	}
+	c.cache.provisionedProductOutputs = getPPOutput.Outputs
+
+	cfStackParameters, err := c.client.GetCloudformationStackParameters(getPPOutput.Outputs)
+	if err != nil {
+		return true, awsclient.Wrap(err, *resp.ProvisionedProductDetail.Name)
+	}
+	if *cr.Spec.ForProvider.ProvisioningArtifactID != *resp.ProvisionedProductDetail.ProvisioningArtifactId ||
+		!provisioningParamsAreNotChanged(cfStackParameters, cr.Spec.ForProvider.ProvisioningParameters) {
+		return false, nil
+	}
+	return true, nil
+}
+
+func (c *custom) lateInitialize(spec *svcapitypes.ProvisionedProductParameters, resp *svcsdk.DescribeProvisionedProductOutput) error {
+	if *resp.ProvisionedProductDetail.LastSuccessfulProvisioningRecordId != "" {
+		describeRecordInput := svcsdk.DescribeRecordInput{Id: resp.ProvisionedProductDetail.LastSuccessfulProvisioningRecordId}
+		describeRecordOutput, err := c.client.DescribeRecord(&describeRecordInput)
+		if err != nil {
+			return err
+		}
+		pathID := describeRecordOutput.RecordDetail.PathId
+		spec.PathID = awsclient.LateInitializeStringPtr(spec.PathID, pathID)
+	}
+	acceptLanguageEnglish := acceptLanguageEnglish
+	if reflect.ValueOf(spec.AcceptLanguage).IsZero() {
+		spec.AcceptLanguage = &acceptLanguageEnglish
+	}
+	return nil
+}
+
+func (c *custom) postObserve(_ context.Context, cr *svcapitypes.ProvisionedProduct, resp *svcsdk.DescribeProvisionedProductOutput, obs managed.ExternalObservation, err error) (managed.ExternalObservation, error) {
+	if err != nil {
+		return managed.ExternalObservation{}, err
+	}
+
+	describeRecordInput := svcsdk.DescribeRecordInput{Id: resp.ProvisionedProductDetail.LastRecordId}
+	describeRecordOutput, err := c.client.DescribeRecord(&describeRecordInput)
+	if err != nil {
+		return managed.ExternalObservation{}, err
+	}
+
+	ppStatus := aws.StringValue(resp.ProvisionedProductDetail.Status)
+	switch {
+	case ppStatus == string(svcapitypes.ProvisionedProductStatus_SDK_AVAILABLE):
+		cr.Status.SetConditions(xpv1.Available())
+	case ppStatus == string(svcapitypes.ProvisionedProductStatus_SDK_UNDER_CHANGE) && *describeRecordOutput.RecordDetail.RecordType == "UPDATE_PROVISIONED_PRODUCT":
+		cr.SetConditions(xpv1.Available().WithMessage("Provisioned Product is updating, availability depends on product"))
+	case ppStatus == string(svcapitypes.ProvisionedProductStatus_SDK_ERROR):
+		cr.Status.SetConditions(xpv1.ReconcileError(errors.New(errProvisionedProductStatusSdkError)))
```

Comment:
It is true if provisioned product has Tainted state, but if it has Error update [is not allowed](https://docs.aws.amazon.com/servicecatalog/latest/userguide/enduser-update.html), so crossplane will never reconcile that resource

Reply:
If you want to prevent an update you can look up the state during `Update` and do nothing if it is not allowed - or you return `ResourceIsUpdToDate: true` in `Observe`. Either way, setting the `ReconcileError` condition here is still not correct and you should use `Unvailable`.